internal/mode/static/nginx/runtime/manager.go

```diff
@@ -189,14 +193,8 @@ func (m *ManagerImpl) GetUpstreams() (ngxclient.Upstreams, error) {
 	return *upstreams, nil
 }
 
-type NewProcessHandlerImpl struct{}
-
-// EnsureNginxRunning ensures NGINX is running by locating the main process.
-func (p *NewProcessHandlerImpl) EnsureNginxRunning(ctx context.Context) error {
-	if _, err := p.FindMainProcess(ctx, pidFileTimeout); err != nil {
-		return fmt.Errorf("failed to find NGINX main process: %w", err)
-	}
-	return nil
+type NewProcessHandlerImpl struct{
+	checkFile func (file string) (string, error)
```

Comment:
instead of this your suggestion : https://github.com/nginxinc/nginx-gateway-fabric/pull/1555#discussion_r1607107777, 

I did this : 

type NewProcessHandlerImpl struct{
	checkFile func (file string) (string, error)
}